infrasim/model.py

```diff
@@ -1523,9 +1534,17 @@ def handle_parms(self):
                     self.__boot_order = "d"
                 else:
                     self.__boot_order = "ncd"
-                self.add_option("-boot {}".format(self.__boot_order))
+                boot_param.append("{}".format(self.__boot_order))
             else:
-                self.add_option("-boot {}".format(self.__boot_order))
+                boot_param += self.__boot_order
+        if self.__boot_menu:
+            boot_param.append(",menu={}".format(self.__boot_menu))
```

Comment:
boot_param.append("menu={}".format(...)), without the leading comma, then use `",".join` in line 1547